### DISCUSSION TITLE
Problem with resumed stream session

### DISCUSSION THREAD

**Discussion by skupi20** [LizardByte/community#327] [Sunshine] [answered]
On latest version i have problem with resumed stream session not all game but some like Fifa 23.

Stream start without problem, but when i suspend stream and later wanna resume it i hear sound of game but see desktop, can move pointer but desktop is unresponsive. At start i thought its something with EA or anticheat, but it work fine before.
I tried many thinks but nothing help. I checked sunshine 0.20 and on that version i can resume Fifa 23 without problem.

**Comment by brycerocky** [comment]
I bisected the nightly builds and traced the culprit CL down to:
https://github.com/LizardByte/Sunshine/commit/dc967ccc7bcac83a7f8249a3d95c219e256603ca

The previous CL in the nightly branch history does not exhibit the issue:
https://github.com/LizardByte/Sunshine/commit/fa7c16bd113166e72bc7e375dfa9dcd48399ae0d

**Reply by brycerocky** [reply]
Following up on this, I commented out the logic in tray_windows::tray_update and the issue disappeared.

**Reply by skupi20** [reply]
Great :) when we can expect the fix in nightly?

**Reply by ReenigneArcher** [reply]
That's a huge commit that can't just be reverted. And commenting out code isn't a fix... So whenever someone determines a proper fix and submits a PR, we will review it.

**Reply by brycerocky** [reply]
Sorry didn’t mean to imply that it was a fix, just that it narrowed down the responsible area to that code. I completely agree that commenting out the section isn’t a viable option.

**Reply by ReenigneArcher** [reply]
Yea, it will be helpful to know where the problematic code lives at least.

**Reply by brycerocky** [reply, marked as answer]
I dug into the code a bit more and started looking at each section of ::tray_update. Oddly it seems that some shellapi methods are triggering the issue. For example, ExtractIconEx causes the issue, while substituting ExtractIconW instead does not. Similarly, InsertMenu and LoadImageA are also causing issues. Not sure what to take away from these finding yet and it's definitely not tractable to replace all methods.

**Reply by ReenigneArcher** [reply]
For now, I think there is enough information to create a bug report about this. Do you mind opening an issue, detailing your findings?
